workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/container/AlertOverlayDialogFactory.kt

```diff
@@ -42,25 +43,45 @@ internal object AlertOverlayDialogFactory : OverlayDialogFactory<AlertOverlay> {
     }
 
     for (button in Button.values()) {
-      rendering.buttons[button]
-        ?.let { name ->
-          alertDialog.setButton(button.toId(), name) { _, _ ->
-            rendering.onEvent(ButtonClicked(button))
-          }
-        }
-        ?: run {
-          alertDialog.getButton(button.toId())
-            ?.visibility = View.INVISIBLE
-        }
+      // Force all the buttons to be created so that we can show
+      // and hide them at will during updates. They get created
+      // when show is called, so we can't mess with them before that
+      // (e.g. to fix the dismiss-on-click nonsense),
+      // and we can't create new ones after that.
+      //
+      // https://github.com/square/workflow-kotlin/issues/138
+      // Why " "? An empty string means no button.
+      alertDialog.setButton(button.toId(), " ") { _, _ -> }
```

Comment:
@christiandeange This the fix for #138, remember that? Can you believe how hacky?

Reply:
I should move this part to `buildDialog` instead of doing it here in `updateDialog`, less confusing.